pkg/envoy/agent.go

```diff
@@ -180,9 +183,17 @@ func (a *Agent) terminate() {
 			default:
 				if err != nil {
 					log.Errorf(err.Error())
-					a.abortCh <- errAbort
-					log.Infof("Graceful termination logic ended prematurely, error while obtaining downstream_cx_active stat")
-					break graceful_loop
+					retryCount++
+					// Max retry 5 times
+					if retryCount > 4 {
+						a.abortCh <- errAbort
+						log.Warnf("Graceful termination logic ended prematurely, error while obtaining downstream_cx_active stat (Max retry %d exceeded)", retryCount)
+						break graceful_loop
+					}
+					log.Warnf("Retrying (%d attempt) to obtain active connections...", retryCount)
+					time.Sleep(retryBackoffTime)
```

Comment:
I don't understand why we are sleeping inside of a ticker loop. Shouldn't we just wait for the next tick?

Sleep here is inappropriate since it won't abort if `statusCh` changes

Reply:
Hmm... I agree with what you said. It's a bit weird to implement exponential backoff retries inside a ticker loop without Replacing the ticker instance dynamically, and it would make the implementation too complicated. I'll see if there's something more appropriate.

Reply:
do we need back off?